### ISSUE TITLE
Display study area and sampling stations for each result

### ISSUE THREAD

**Issue by reedhewitt** [GoMexSI/gomexsi-web#94]
Jorrit, is there a way to update the query results so that they return the study area polygon and sampling stations, if applicable?  Currently we only have the coordinates of the centroid.  I'll update the interface so that the additional data shows on the individual map for each result.

**Comment by jhpoelen** [comment]
@reedhewitt I was hoping that we'd get to this point! I've implemented the first phase of study area polygon. Next step is for GloBI to populate and expose the polygons. 

I was imagining to have something like:

http://api.globalbioticinteractions.org/locations?field=latitude&field=longitude&field=footprintWKT&accordingTo=gomexsi 

This way, you can request the footprints while keeping the option of just getting the (lightweight) centroids.

Curious to hear your ideas on this!

**Comment by reedhewitt** [comment]
@jhpoelen Just got your email and realized I never replied to this.  In the link you sent, I see an array of lat/lng pairs, but I don't see anything that appears to describe a polygon.  I must be missing something.  Where in that data would I find the footprints for the study areas?

**Comment by jhpoelen** [comment]
Yep, lat/lng pairs is current behavior. With proposed (not yet implemented behavior), a third column would be present with name "footprintWKT". Before implementing this, I wanted to make sure that this would suite your use case. If so, please let me know and I can work on it.

**Comment by reedhewitt** [comment]
Ah, I get it. Yes, that would be perfect.  Sorry for the delayed response.

**Comment by jhpoelen** [comment]
Great! Will let you know when it is available.

**Comment by jhpoelen** [comment]
hey @reedhewitt 
http://api.globalbioticinteractions.org/locations?accordingTo=gomexsi now returns footprintWKT values. Please confirm that this contains the information you need by closing this issue. Thanks for being patient!

**Comment by reedhewitt** [comment]
Thanks, Jorrit!  I'll take a look and close this ticket if I don't have any questions.

**Comment by reedhewitt** [comment]
@jhpoelen I've come across an obstacle. The front end gets the query result instance data from a request handler on the TAMUCC server (beginning at gomexsi/requestHandler/RequestHandler.php). The request handler gets the data from GloBI, formats it into a result object, and passes it on to the front end as JSON.

The request handler does not include the newly available footprintWKT values in the response object. I didn't work on the request handler code at all, and I'm having a hard time finding where to modify the response object.

Can you point me in the right direction or update the response object? (Whichever is faster.)

**Comment by reedhewitt** [comment]
I'm testing with data copied directly from http://api.globalbioticinteractions.org/locations?accordingTo=gomexsi until we get it in the query response object.  I noticed that the polygon strings can't be parsed, which seems to be a result of extra parentheses around each lat/lng pair.

The current data looks like this:
POLYGON((-95.58771174 25.9127210949999),(-96.3584874496999 25.9357004433999),(-96.3458669097999 25.9802162195),(-96.3422894947999 26.028444829), ... )

The WKT Wikipedia page (https://en.wikipedia.org/wiki/Well-known_text#Geometric_objects) indicates it should like like this:
POLYGON((-95.58771174 25.9127210949999, -96.3584874496999 25.9357004433999, -96.3458669097999 25.9802162195, -96.3422894947999 26.028444829, ... ))

There should be double parentheses at the start and end, but none within the list of lat/lng pairs.  I'm using Wicket (http://arthur-e.github.io/Wicket/sandbox-gmaps3.html) to display the WKT polygons on the instance maps.

**Comment by jhpoelen** [comment]
Thanks for the comments. It seems that we are dealing with two issues (1) how to get the footprintWKT and (2) format of the footprintWKT . 

Related to (1), I'd say, if it is easier to bypass the request handler, let's remove the intermediate request handler code. If not, I'd be happy to help make changes necessary. 

Related to (2): the footprintWKT: I was assuming that the values in GoMexSI locations.csv / LOC_POLY_COORDS column where wkt formatted. As you noticed . . . they are currently not. I'll work on fixing that shortly.

Please let me know your thoughts on (1).

**Comment by jhpoelen** [comment]
Found and fixed the issue related to (2). Thanks for noticing that! The changes should propagate within a couple of days.

**Comment by reedhewitt** [comment]
Thanks for fixing (2).  Regarding (1), I'm fine with bypassing the intermediate request handler, but I'll need some help knowing what GloBI endpoint to use.  For example, a taxonomic query for Ariopsis felis currently sends this query string to the request handler via post:

subjectName=Ariopsis+felis&allInteractions=on&findPrey=on&filterPrey=&findPredators=on&filterPredators=&serviceType=rest

What comes back from the request handler is the JSON that you can see here: http://pastebin.com/2eMgVTvU

That JSON is perfect, except that each preyInstance and predInstance needs to have a polygon property added.  Is it possible to get that JSON response directly from the GloBI API?

**Comment by jhpoelen** [comment]
Ok. Sounds like it might be easier to stick with current approach and add footprintWKT to results. Working on it...

**Comment by jhpoelen** [comment]
@reedhewitt I took a first pass at addressing (1). Please review my changes and deploy when you feel they are sufficient. Note that I code somewhat defensively to not introduce too many changes.

Hope this helps, and please let me know if you have any questions. 

ps, on the server, I noticed that there's some local changes, which I am assuming are made by you. This is why I was hesitant to deploy the changes myself. 

```
gomexsi@falcon1 html]$ cd gomexsi
[gomexsi@falcon1 gomexsi]$ git status
# On branch master
# Your branch is ahead of 'origin/master' by 2 commits.
#
# Changed but not updated:
#   (use "git add <file>..." to update what will be committed)
#   (use "git checkout -- <file>..." to discard changes in working directory)
#
#	modified:   gomexsi-wp/functions.php
#	modified:   gomexsi-wp/style.css
#
no changes added to commit (use "git add" and/or "git commit -a")
[gomexsi@falcon1 gomexsi]$ 
```

with local git commit log 
```
commit bf1955c84e1f57eecc28a48d5e0cdd603ecf4c82
Author: gomexsi <gomexsi@falcon1.tamucc.edu>
Date:   Fri Jan 22 14:10:41 2016 -0600

    move registration form to its own page

commit 3127ad17e5dc99970e61b7867e521361eab2e869
Author: gomexsi <gomexsi@falcon1.tamucc.edu>
Date:   Wed Jan 20 10:35:01 2016 -0600

    cleanup javascript for stat counter
```

**Comment by reedhewitt** [comment]
Yeah, sorry about the non-committed local changes.  I accidentally hit "upload" on some local files I had changed.  I didn't bother reversing the change at the time because my edits were inconsequential and I thought I was about to launch the other changes I have been making.  It has all taken longer than planned.

I'll deploy your changes and verify that everything is good with my previous changes.

**Comment by jhpoelen** [comment]
Sounds good. Please close this issue whenever you have what you need . . .

**Comment by reedhewitt** [comment]
I have merged my local changes with your changes to the repo and I have pulled the updates to Falcon. I can confirm that the WKT footprint is showing up in the results data. Once the change to remove the extra parentheses propagates, we'll be all set. Thanks!

**Comment by jhpoelen** [comment]
@reedhewitt I just checked GloBI, and the polygons look much better now. Please confirm.

**Comment by reedhewitt** [comment]
@jhpoelen Yes, the formatting is perfect and the polygons are readable by the WKT library I'm using.  Do you want to close this issue since the change is live in GloBI, or wait until the interface has been updated on GoMexSI?

<img width="952" alt="study footprint" src="https://cloud.githubusercontent.com/assets/957141/14081243/5b35326e-f4cf-11e5-9a52-5139cedb56d8.png">

**Comment by jhpoelen** [comment]
@reedhewitt looks pretty cool! Nice work! 

I'd like to suggest to re-open the issue and mark it with "ready for deployment". This way, we can easily see which feature are ready for the next release. If you have other ideas, please let me know.

**Comment by reedhewitt** [comment]
Works for me.
